### PR TITLE
EP-444: CTA Clicked (search) Unit Test

### PR DESCRIPTION
**Pull request by sunday-okpoluaefe**
# 📲 What

Android: CTA Clicked (search) Unit Test

# 🤔 Why

Test for Context Properties, Segment Properties, Session Properties, User Properties

# 🛠 How

```
 @Test
    fun testSearchCta_Properties() {
        val user = user()
        val client = client(user)
        client.eventNames.subscribe(this.segmentTrack)
        client.eventProperties.subscribe(this.propertiesTest)
        client.identifiedId.subscribe(this.segmentIdentify)
        val segment = AnalyticEvents(listOf(client))

        val params = DiscoveryParams
            .builder()
            .category(CategoryFactory.ceramicsCategory())
            .sort(DiscoveryParams.Sort.MAGIC)
            .build()

        segment.trackSearchCTAButtonClicked(params)

        assertSessionProperties(user)
        assertContextProperties()
        assertUserProperties(false)

        val expectedProperties = propertiesTest.value
        assertEquals(EventContextValues.CtaContextName.SEARCH.contextName, expectedProperties[CONTEXT_CTA.contextName])
        assertEquals(EventContextValues.LocationContextName.GLOBAL_NAV.contextName, expectedProperties[ContextPropertyKeyName.CONTEXT_LOCATION.contextName])

        this.segmentTrack.assertValue(EventName.CTA_CLICKED.eventName)
    }
```

# 👀 See

Trello, screenshots, external resources?

| Before 🐛 | After 🦋 |
| --- | --- |
|  |  |

# 📋 QA

Instructions for anyone to be able to QA this work.

# Story 📖

https://kickstarter.atlassian.net/browse/EP-444

### CODE REVIEW

File: app/src/test/java/com/kickstarter/libs/SegmentTest.kt [hunk 1]
```diff
@@ -242,6 +243,34 @@ class SegmentTest : KSRobolectricTestCase() {
         this.segmentIdentify.assertValue(user.id())
     }
 
+    @Test
+    fun testSearchCta_Properties() {
```
Comment (leighdouglas) [inline app/src/test/java/com/kickstarter/libs/SegmentTest.kt:247]: Does this search CTA event also send discover properties? if so, we also need to test for those as well.

**Review by leighdouglas — CHANGES_REQUESTED** [review]
If search CTA sends discover properties, we should be testing for those as well.

File: app/src/test/java/com/kickstarter/libs/SegmentTest.kt (hunk 1 above)
Comment (sunday-okpoluaefe) [reply app/src/test/java/com/kickstarter/libs/SegmentTest.kt:247]: Actually, it does not. I am refactoring the origin ticket. thanks

**Review by leighdouglas — APPROVED** [review]
LGTM! 🎉

**Pull request merged**